### PR TITLE
Update `Job` related commands to use `Index`

### LINKED ISSUE #96: `Assign` command `assign j/2 p/1 p/2 p/4`
based on the 2 lists being displayed
- stores the `Rate basePay` of the persons at the time of assignment of the job
- [so that editing the person's `Rate basePay` after the assignment doesn't change their previous pay]

### PR DESCRIPTION
**Pull request by Spyobird**
Command usage is:
`assign JOB_INDEX i/PERSON_INDEX...`

Requires #106 to be merged.

Closes #96.

Also added `JobFindCommand`.

### CODE REVIEW

File: src/main/java/peoplesoft/commons/core/index/Index.java [hunk 1]
```diff
@@ -9,6 +9,8 @@
  * convert it back to an int if the index will not be passed to a different component again.
  */
 public class Index {
+    public static final String MESSAGE_CONSTRAINTS = "Index should be a non-zero positive integer.";
+
```
Comment (ian-from-dover) [inline src/main/java/peoplesoft/commons/core/index/Index.java:13]: If the user will see this message, I really think HR managers want to put this kinda lingo behind them 😩 If not, it is fine.

File: src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java [hunk 2]
```diff
@@ -1,102 +1,128 @@
...
 
-    private ID jobId;
-    private Index personIndex;
+    private Index jobIndex;
+    private Set<Index> personIndexes;
+    private Employment instance;
 
     /**
-     * Creates a {@code JobAssignCommand} to assign a {@code Job} to a {@code Person}.
+     * Creates a {@code JobAssignCommand} to assign a {@code Job} to a set of {@code Persons}.
```
Comment (zhongfu) [inline src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java:43]: minor style nitpick
```suggestion
     * Creates a {@code JobAssignCommand} to assign a {@code Job} to a set of {@code Person}s.
```

File: src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java [hunk 3]
```diff
@@ -1,102 +1,128 @@
...
-            + PREFIX_INDEX + "INDEX ";
+            + "JOB_INDEX "
+            + PREFIX_INDEX + "PERSON_INDEX [PERSON_INDEX]...";
 
-    public static final String MESSAGE_SUCCESS = "Assigned Job %s to %s\n%s has the following jobs: %s";
-    public static final String MESSAGE_JOB_NOT_FOUND = "This job does not exist";
-    public static final String MESSAGE_DUPLICATE_EMPLOYMENT = "This person is already assigned to this job";
+    public static final String MESSAGE_SUCCESS = "Assigned Job %s to %s.";
+    public static final String MESSAGE_DUPLICATE_EMPLOYMENT =
+            "A total of %s persons have existing employments.";
```
Comment (zhongfu) [inline src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java:36]: "MESSAGE_DUPLICATE_EMPLOYMENT"?

and also i think the error msg is probably not too user friendly, "employments" is a rather weird term to use

Comment (zhongfu) [reply src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java:36]: oh yeah ok i get what you meant by that

Comment (ian-from-dover) [inline src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java:36]: "%s people have been assigned to jobs" may sound better

File: src/main/java/peoplesoft/logic/parser/FindCommandParser.java [hunk 4]
```diff
@@ -19,10 +19,12 @@
...
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
+        String trimmedArgs;
+        try {
+            trimmedArgs = ParserUtil.parseString(args);
+        } catch (ParseException pe) {
             throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE), pe);
```
Comment (ian-from-dover) [inline src/main/java/peoplesoft/logic/parser/FindCommandParser.java:27]: Indent should be 2 tabs here

File: src/test/java/peoplesoft/logic/parser/job/JobAssignCommandParserTest.java [hunk 5]
```diff
@@ -1,23 +1,71 @@
...
+    public void parse_validArgs_success() {
+        // One index for person
+        JobAssignCommand expected = new JobAssignCommand(Index.fromOneBased(3),
+                new HashSet<Index>(Arrays.asList(Index.fromOneBased(2))), new Employment());
+        assertParseSuccess(parser, "3 i/2", expected);
+
+        // Multiple indexes for person
+        expected = new JobAssignCommand(Index.fromOneBased(2), new HashSet<Index>(Arrays.asList(
+                Index.fromOneBased(1), Index.fromOneBased(3), Index.fromOneBased(4))), new Employment());
+        assertParseSuccess(parser, "2 i/1 i/4 i/3", expected);
```
Comment (ian-from-dover) [inline src/test/java/peoplesoft/logic/parser/job/JobAssignCommandParserTest.java:69]: nice 👍

**Review by ian-from-dover — APPROVED** [review]
LGTM, just a few idnent things

File: src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java (hunk 3 above)
Comment (ian-from-dover) [reply src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java:36]: Perhaps "A total of %s people have jobs assigned to them." may work better

Comment (Spyobird) [reply src/main/java/peoplesoft/logic/commands/job/JobAssignCommand.java]: Changed to "%s person(s) have already been assigned to this job."

**Comment by ian-from-dover** [comment]
sike i beat you to it

**Pull request merged**